radixdlt-java/src/main/java/com/radixdlt/client/core/identity/PrivateKeyEncrypter.java

```diff
@@ -96,7 +100,8 @@ public static void createEncryptedPrivateKeyFile(String password, String filePat
         byte[] computedMac = generateMac(derivedKey.getEncoded(), cipherText);
 
         if (!Arrays.equals(computedMac, ByteString.decodeHex(mac).toByteArray())) {
-            throw new Exception("MAC mismatch");
+            // TODO: Change to MacMismatchException and make that exception extend GeneralSecurityException?
```

Comment:
The crypto package already has a MacMismatchException which you can use though it extends the Exception class but you are right, it would make sense to throw this kind of exception so you can go ahead and make this change IMO and remove the TODO